components/Chips/src/MDCChipView.h

```diff
@@ -0,0 +1,190 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/*
+ A Material chip.
+
+ @see https://material.io/guidelines/components/chips.html for full details.
+
+ Chips are compact elements that represent an attribute, text, entity, or action.
+
+ Chips contain an optional leading image, a title label and an optional trailing accessory view.
+ They can also contain a leading image that appears only when the chip is selected.
+ */
+@interface MDCChipView : UIControl
+
+/*
+ A UIImageView that leads the title label.
+
+ Only visible when the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) IBInspectable UIImageView *imageView;
+
+/*
+ A UIImageView that leads the title label. Appears above the imageView.
+
+ This image view is typically used to show some icon that denotes the chip as selected, such as a
+ tick. If imageView has no image then the chip will require resizing when selected or deselected to
+ account for the changing visibility of selectedImageView.
+
+ Only visible when the chip is selected and the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) IBInspectable UIImageView *selectedImageView;
+
+/*
+ A UIView that trails the title label.
+
+ It will be given a size based on the value returned from sizeThatFits:.
```

Comment:
Is the chip view's sizeThatFits: a function of the accessoryView's sizeThatFits? E.g. if accessoryView has a size of 100x100, will the chip's also be at least 100x100?

If not, how is the accessory view's sizeThatFits result taken into account? This may be beneficial to document here as an expansion of the current statement.

Reply:
The chips sizeThatFits is a function of all of its subviews sizeThatFits.